### PR TITLE
FCT-457: Transaction and Transaction Draft Models 

### PR DESCRIPTION
**Pull request by tylermorrisford**
This PR adds the `Transaction` and `TransactionDraft` models.

### CODE REVIEW

File: models/payment/src/transaction/generator.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const [getOlderDate] = createRelatedDates();
+
+// https://docs.commercetools.com/api/projects/payments#transaction
+
+const generator = Generator<TTransaction>({
+  fields: {
+    id: fake((f) => f.string.uuid()),
+    timestamp: fake(getOlderDate),
+    type: oneOf(...Object.values(TRANSACTION_TYPE)),
+    amount: fake((f) => Money.random()),
```
Comment (jaikamat) [inline models/payment/src/transaction/generator.ts:16]: I believe this should be `CentPrecisionMoney`

File: models/payment/src/transaction/generator.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    interactionId: null,
+    state: oneOf(...Object.values(TRANSACTION_STATE)),
+    custom: null,
```
Comment (jaikamat) [inline models/payment/src/transaction/generator.ts:19]: AIN'T NOBODY GOT TIME FOR THIS 👏

File: models/payment/src/transaction/transaction-draft/generator.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const [getOlderDate] = createRelatedDates();
+
+// https://docs.commercetools.com/api/projects/payments#transactiondraft
+
+const generator = Generator<TTransactionDraft>({
+  fields: {
+    timestamp: fake(getOlderDate),
+    type: oneOf(...Object.values(TRANSACTION_TYPE)),
+    amount: fake((f) => Money.random()),
```
Comment (jaikamat) [inline models/payment/src/transaction/transaction-draft/generator.ts:15]: Same here, `CentPrecisionMoney`?

File: models/payment/src/transaction/generator.ts (hunk 1 above)
Comment (tylermorrisford) [reply models/payment/src/transaction/generator.ts:16]: bah - you are correct! updating

File: models/payment/src/transaction/generator.ts (hunk 2 above)
Comment (tylermorrisford) [reply models/payment/src/transaction/generator.ts:19]: ... a pattern I saw and loved!

**Review by jaikamat — APPROVED** [review]
Looks good to me. Glad the GraphQL entities didn't deviate from REST too much!

File: models/payment/src/transaction/transaction-draft/generator.ts (hunk 3 above)
Comment (tylermorrisford) [reply models/payment/src/transaction/transaction-draft/generator.ts]: Oops - I changed it and then noticed that this one should actually be the `Money` model...

Comment (jaikamat) [reply models/payment/src/transaction/transaction-draft/generator.ts:15]: Good catch! `Money` it is

Comment (jaikamat) [reply models/payment/src/transaction/transaction-draft/generator.ts:15]: Huh so I wonder if the pattern is `CentPrecisionMoney` for representation, `Money` for draft

**Pull request merged**